torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -582,6 +583,7 @@ bool ProcessGroupNCCL::WorkNCCL::finishedGPUExecutionInternal() const {
 
 bool ProcessGroupNCCL::WorkNCCL::checkTimeout(
     std::optional<std::chrono::milliseconds> timeout) {
+  SCOPED_WAIT_COUNTER_US("pytorch.logging.wait_counter.ProcessGroupNCCL::WorkNCCL::checkTimeout");
```

Comment:
would it be convention that all counters have the same prefix "pytorch.logging.wait_counter"?  If so maybe its worth prepending that inside the counter impl and then users can just write the suffix "ProcessGroup..."

Reply:
I think that for these wait counters, we would have the same prefix pytorch.wait_counter (i took out the logging).

Reply:
the downside of having the prefix put in implicitly, is that it might make the counters less searchable in the code?